File: 1951-FindTheWinnerOfTheCircularGame/1951-FindTheWinnerOfTheCircularGame.py
# Last updated: 8/9/2025, 1:39:42 PM
class Solution(object):
    def findTheWinner(self, n, k):
        l= list(range(1, n + 1))
        start=0
        while len(l) > 1:
            remove= (start+k-1) % len(l)
            l.pop(remove)

            start=remove

        return l[0]
        

File: 1951-FindTheWinnerOfTheCircularGame/1951-FindTheWinnerOfTheCircularGame.java
// Last updated: 8/9/2025, 1:39:36 PM
class Solution {
    public int findTheWinner(int n, int k) {
        List<Integer> ll = new ArrayList<>();
        for(int i=1;i<=n;i++){
            ll.add(i);
        }
        int idx=0;
        while(ll.size()>1){
            idx = (idx+k-1)% ll.size();
            ll.remove(idx);
        }
        return ll.get(0);
        
    }
}